src/common/telemetry/collect-telemetry/collect-telemetry.ts

```diff
@@ -0,0 +1,149 @@
+import {execSync} from 'child_process'
+import * as fs from 'fs'
+import fetch from 'node-fetch'
+import * as path from 'path'
+import {ProjenrcFile} from 'projen'
+import type {NodeProject, NodeProjectOptions} from 'projen/lib/javascript'
+import type {IWithTelemetryReportUrl} from '../with-telemetry'
+import {cloneWorkflow} from './clone-workflow'
+import {diff} from './diff'
+import {EscapeHatches, getEscapeHatches} from './get-escape-hatches'
+
+interface TelemetryPayload {
+  version?: string
+  gitUrls?: Array<string>
+  escapeHatches?: {
+    files?: EscapeHatches
+    overrides?: EscapeHatches
+    tasks?: EscapeHatches
+  }
+  workflows?: {
+    deleted?: Array<string>
+    new?: Array<object>
+    updated?: Array<object>
+  }
+  errors?: unknown
+}
+
+export const telemetryEnableEnvVar = 'IS_OTTOFELLER_TEMPLATES_TELEMETRY_COLLECTED' as const
+
+export const collectTelemetry = async (project: NodeProject & IWithTelemetryReportUrl) => {
+  if (!project.telemetryReportUrl || !process.env[telemetryEnableEnvVar]) {
+    return
+  }
+
+  project.logger.info('Collect project telemetry')
+  const payload: TelemetryPayload = {}
+  const errors: Array<unknown> = []
+
+  // ANCHOR template version
+  try {
+    const version = project.package.tryResolveDependencyVersion('@ottofeller/templates')
+
+    if (version) {
+      payload.version = version
+    }
+  } catch (e) {
+    errors.push(e)
+  }
+
+  // ANCHOR git urls
```

Comment:
@quesabe why is it plural? Is it because we collect origins URLs here?

Reply:
Yes. Git allows multiple remotes, so we pull them all.